apps/ejabberd/src/mongoose_metrics_hooks.erl

```diff
@@ -110,64 +110,75 @@ get_hooks(Host) ->
-    mongoose_metrics:update(Server, sessionAuthFails, 1).
+-spec auth_failed(map(), binary(), binary()) -> metrics_notify_return().
+auth_failed(Acc, _,Server) ->
+    mongoose_metrics:update(Server, sessionAuthFails, 1),
+    Acc.
 
--spec user_send_packet(ejabberd:jid(), tuple(), tuple()
+-spec user_send_packet(map(), ejabberd:jid(), tuple(), tuple()
                       ) -> metrics_notify_return().
-user_send_packet(#jid{server = Server},_,Packet) ->
+user_send_packet(Acc, #jid{server = Server},_,Packet) ->
     mongoose_metrics:update(Server, xmppStanzaSent, 1),
-    user_send_packet_type(Server, Packet).
+    user_send_packet_type(Server, Packet),
+    Acc.
 
 -spec user_send_packet_type(Server :: ejabberd:server(),
-                            Packet :: jlib:xmlel()) -> metrics_notify_return().
+                            Packet :: jlib:xmlel()) -> ok | {error, not_found}.
 user_send_packet_type(Server, #xmlel{name = <<"message">>}) ->
     mongoose_metrics:update(Server, xmppMessageSent, 1);
 user_send_packet_type(Server, #xmlel{name = <<"iq">>}) ->
     mongoose_metrics:update(Server, xmppIqSent, 1);
 user_send_packet_type(Server, #xmlel{name = <<"presence">>}) ->
     mongoose_metrics:update(Server, xmppPresenceSent, 1).
 
--spec user_receive_packet(ejabberd:jid(), tuple(), tuple(), tuple()) -> term().
-user_receive_packet(#jid{server = Server} ,_,_,Packet) ->
+-spec user_receive_packet(map(), ejabberd:jid(), tuple(), tuple(), tuple()) -> term().
+user_receive_packet(Acc, #jid{server = Server} ,_,_,Packet) ->
     mongoose_metrics:update(Server, xmppStanzaReceived, 1),
-    user_receive_packet_type(Server, Packet).
+    user_receive_packet_type(Server, Packet),
+    Acc.
 
 -spec user_receive_packet_type(Server :: ejabberd:server(),
-                               Packet :: jlib:xmlel()) -> metrics_notify_return().
+                               Packet :: jlib:xmlel()) -> ok | {error, not_found}.
 user_receive_packet_type(Server, #xmlel{name = <<"message">>}) ->
     mongoose_metrics:update(Server, xmppMessageReceived, 1);
 user_receive_packet_type(Server, #xmlel{name = <<"iq">>}) ->
     mongoose_metrics:update(Server, xmppIqReceived, 1);
 user_receive_packet_type(Server, #xmlel{name = <<"presence">>}) ->
     mongoose_metrics:update(Server, xmppPresenceReceived, 1).
 
--spec xmpp_bounce_message(Server :: ejabberd:server(),
+-spec xmpp_bounce_message(Acc :: map(), Server :: ejabberd:server(),
                           tuple()) -> metrics_notify_return().
-xmpp_bounce_message(Server, _) ->
-    mongoose_metrics:update(Server, xmppMessageBounced, 1).
-
--spec xmpp_stanza_dropped(ejabberd:jid(), tuple(), tuple()) -> metrics_notify_return().
-xmpp_stanza_dropped(#jid{server = Server} ,_,_) ->
-   mongoose_metrics:update(Server, xmppStanzaDropped, 1).
-
--spec xmpp_send_element(Server :: ejabberd:server(),
+xmpp_bounce_message(Acc, Server, _) ->
+    case mongoose_metrics:update(Server, xmppMessageBounced, 1) of
```

Comment:
Why do we want to modify Acc when metric update fails? The same applies to the one below.